WebApplicationDemo/Services/PostService.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Flurl.Http.Configuration;
+using Flurl.Http;
+using System.Collections.Generic;
+
+using WebApplicationDemo.Application.Handlers.Queries;
+using WebApplicationDemo.Dto;
+using System.Threading.Tasks;
+using System.Text.Json;
+using System.Threading;
+using System;
+
+namespace WebApplicationDemo.Services
+{
+    public sealed class PostService : IPostService, IDisposable
+    {
+        private readonly IFlurlClientCache _clientCache;
+        private readonly IFlurlClient _jsonPlaceHolderClient;
+
+        private string _url = "https://jsonplaceholder.typicode.com/";
+
+        public PostService(IFlurlClientCache clientCache)
+        {
+            _clientCache = clientCache;
+
+            _jsonPlaceHolderClient = _clientCache.GetOrAdd(typeof(PostService).FullName, _url, builder =>
+            {
+                // customize client
+                builder.WithTimeout(10);
+            });
+        }
+
+        public async Task<List<Post>> GetPosts(CancellationToken ctk)
+        {
+            var response = await _jsonPlaceHolderClient.Request("posts").GetStringAsync(cancellationToken: ctk);
+
+            var data = JsonSerializer.Deserialize<List<Post>>(response);
+
+            return data ?? new List<Post>();
+        }
+
+        public void Dispose()
+        {
+            _clientCache.Remove(typeof(PostService).FullName);
```

Comment:
Hmm, I'm not sure what the 'cache' gives us in our scenario. Looks only as a soirce of issues.

Why not doing just 

``` 
_client = new FlurlClient(_url).ConfigureForArk(); 
```

I mean, using a cache doesn't add any value as far as I can see. Only risk of not Removing it. 
And if Service is for any reason not a Singleton, we also misbehave on concurrent requests.

What are the benefits?

Reply:
Well from my understanding a benefit would be if we're using the cache in a transient or scoped service, we would not have to reinstantiate and dispose the client on each request. As far as disposing the cache I thought we could dispose it on app service shut down.

Reply:
Right but at that point you should not Remove it from the Cache. But if you don't, you're "ignoring" a IDisposable object returned which is a bad pattern.

What I'm saying is that FlurlCache gives us only headache (shall I dispose it?, shall I remove it? where I shall configure the Client as I need it?) because ismt defines a poor interface (should have returned a Wrapped Client which can be Disposed safely without breaking the Cache).

Give the poor interface, I'd be inclined not to use it than to keep it. Not Disposing a returned Disposable is bad, doing so breaks the contract of the Cache.